drivers/sgp30/sgp30.c

```diff
@@ -229,14 +247,13 @@ int sgp30_read_serial_number(sgp30_t *dev, uint8_t *buf, size_t len)
         DEBUG_PUTS("[sgp30]: fail read");
         return -EPROTO;
     }
-    if (_crc8(frame, 2) != frame[2] || _crc8(frame + 3, 2) != frame[5] ||
-        _crc8(frame + 6, 2) != frame[8]) {
+    /* the serial id is in big endian format */
+    if (_get_uint16_and_check_crc(&frame[0], (uint16_t *)&buf[4]) ||
```

Comment:
this is now possibly causing alignment errors, we cannot cast any buf to `uint16_t`.

Reply:
I suggest using a temp `uint16_t[3]` and memcpy'ing it.